Svc/GroundInterface/GroundInterface.cpp

```diff
@@ -125,8 +125,23 @@ namespace Svc {
     routeComData()
   {
       // Read the packet type from the data buffer
-      U32 packet_type = Fw::ComPacket::FW_PACKET_UNKNOWN;
-      m_in_ring.peek(packet_type, HEADER_SIZE);
+      FwPacketDescriptorType packet_type = Fw::ComPacket::FW_PACKET_UNKNOWN;
+
+      //read packet descriptor in size agnostic way
+      U8 packet_descriptor_size = sizeof(FwPacketDescriptorType);
+      U8 packet_type_bytes[packet_descriptor_size];
+      Fw::SerializeStatus stat = m_in_ring.peek(packet_type_bytes, packet_descriptor_size, HEADER_SIZE);
+      //m_in_ring.peek(packet_type, HEADER_SIZE); // this way is only valid for 4byte packet descriptors
+      if(stat == Fw::FW_SERIALIZE_OK)
```

Comment:
I'd recommend we refactor this to use our standard serialization code.  This isn't a blocker.

Reply:
Based on the review comment this would involve adding peel methods for U16 and U64 types. Are there any other types that might be needed? Like signed types perhaps?